lib/gush/worker.rb

```diff
@@ -51,7 +51,17 @@ def setup_job(workflow_id, job_id)
     def incoming_payloads
       payloads = {}
       job.incoming.each do |job_name|
-        payloads[job_name] = client.load_job(workflow.id, job_name).output_payload
+       job = client.load_job(workflow.id, job_name)
+       payloads[job.class.to_s] ||= []
+
+       if workflow.nameize_payloads
+         payload = {:id => job.name, :payload => job.output_payload}
```

Comment:
Yeah that sounds good. Just extracting `:payload` values will most likely be the thing developers use, but internally we can store hashes.